server/athenian/api/align/queries/metrics.py

```diff
@@ -194,6 +203,62 @@ def _parse_time_interval(params: Mapping[str, Any]) -> Intervals:
     return goal_dates_to_datetimes(date_from, date_to)
 
 
+def _parse_requested_metrics(params: Mapping[str, Any]) -> Sequence[_RequestedMetric]:
+    metrics = params.get(MetricParamsFields.metrics)
+    metrics_w_params = params.get(MetricParamsFields.metricsWithParams)
+
+    if not (bool(metrics) ^ bool(metrics_w_params)):
+        fields = [MetricParamsFields.metrics, MetricParamsFields.metricsWithParams]
+        raise ResponseError(
+            InvalidRequestError(
+                pointer=" or ".join(f".{f}" for f in fields), detail=f"Use {' or '.join(fields)}",
+            ),
+        )
+
+    if metrics:
+        return [_RequestedMetric(m, None, None) for m in metrics]
+
+    assert metrics_w_params
+    req_metrics = []
+    for m_w_params in metrics_w_params:
+        name = m_w_params[MetricWithParamsInputFields.name]
+        metric_params = convert_metric_params_datatypes(
+            parse_metric_params(m_w_params.get(MetricWithParamsInputFields.metricParams)),
+        )
+        raw_teams_params = m_w_params.get(MetricWithParamsInputFields.teamsMetricParams) or ()
```

Comment:
```suggestion
        raw_teams_params = m_w_params.get(MetricWithParamsInputFields.teamsMetricParams, ())
```

Reply:
I had to check this manually,
in GraphQL null and optional are the same, if client passes `teamsMetricParams: null` we receive the key with None value

Reply:
Ah so `.get()` returns `None`, OK